source/MaterialXGenGlsl/Nodes/BlurGlsl.cpp

```diff
@@ -65,57 +69,124 @@ void BlurGlsl::emitFunctionCall(const SgNode& node, SgNodeContext& context, Shad
         _inputTypeString = shadergen.getSyntax()->getTypeName(inInput->type);
     }
 
-    const SgInput* filterTypeInput = node.getInput("filtertype");
+    const string FILTER_TYPE_STRING("filtertype");
+    const SgInput* filterTypeInput = node.getInput(FILTER_TYPE_STRING);
     if (!inInput || !filterTypeInput || _inputTypeString.empty())
     {
         throw ExceptionShaderGenError("Node '" + node.getName() + "' is not a valid Blur node");
     }
 
-    // Check for type of filter to apply
-    //
-    const string BOX_FILTER("box");
-    const string GAUSSIAN_FILTER("gaussian");
-    if (filterTypeInput->value)
+    // Compute width of filter. Default is 1 which just means one 1x1 upstream samples
+    const string FILTER_SIZE_STRING("size");
+    const SgInput* sizeInput = node.getInput(FILTER_SIZE_STRING);
+    _filterWidth = 1;
+    if (sizeInput)
     {
-        if (filterTypeInput->value->getValueString() == GAUSSIAN_FILTER)
+        float sizeInputValue = sizeInput->value->asA<float>();
```

Comment:
Map to discrete filter width sizes. 49 samples pretty well covers the entire area of a Gaussian.